File: src/Views/SignInView.cs
using CineComplex.Classes;
using CineComplex.Classes.Base;
using CineComplex.Interfaces;
using CineComplex.Models;
using CineComplex.Services;
using CineComplex.ViewModels;
using CineComplex.Views.AdminClient;
using CineComplex.Views.FormViews;

namespace CineComplex.Views
{
    public class SignInView : AViewBase<SignInView>, IView
    {
        private int _choice = 0;
        public int Choice { get => _choice; set => _choice = value; }

        public List<string> MenuList
        {
            get;
            set;
        }
        public void View()
        {
            SignInView.Instance.LoadMenuList();

            do//main loop
            {
                Console.Clear();
                Console.WriteLine("\t----- !!! Salam Hindusthan !!! -----");
                Console.WriteLine("================================================");

                Console.WriteLine("\nSignIn View - CineComplex");
                Console.WriteLine("-------------------------------------------------");

                Console.WriteLine($"Entered Login Id : {Credential.Instance.LoginId}");
                Console.WriteLine($"Entered Password : {Credential.Instance.Password}");

                Console.WriteLine();

                Console.WriteLine("\nMenu : ");
                Console.WriteLine("---------------");

                foreach (string instr in Instance.MenuList)
                {
                    Console.WriteLine(instr);
                }

                Console.Write("Enter Your Choice : ");

                int.TryParse(Console.ReadLine(), out _choice);

                switch (Choice)
                {
                    case 1:
                        Console.Write("Enter LoginID : ");
                        SignInViewModel.Instance.GetSignInId();
                        Console.Clear();
                        break;

                    case 2:
                        Console.Write("Enter Password : ");
                        SignInViewModel.Instance.GetSignIPassword();
                        Console.Clear();
                        break;

                    case 3:
                        Result<bool> authenticationResult = SignInViewModel.Instance.SignIn();
                        if (authenticationResult.IsSuccessful)
                        {
                            AdminHomeView.Instance.View();
                        }
                        else
                        {
                            Console.Write($"{authenticationResult.Message}");
                            Console.ReadLine();
                        }
                        break;

                    case 4:
                        SignUpFormView.Instance.View();
                        break;

                    case 5:
                        ForgotPasswordFormView.Instance.View();
                        Console.Clear();
                        break;

                    case 6:
                        SignInViewModel.Instance.ResetFormCommand();
                        break;

                    default:
                        Console.WriteLine("Please enter the valid Choice .....");
                        break;
                }
            } while (Choice != Instance.MenuList.Count);
        }

        public void LoadMenuList()
        {
            Instance.MenuList = new List<string>() {
                "1. Enter Login Id ",
                "2. Enter Password ",
                "3. Login ",
                "4. Sign Up ",
                "5. Forgot Password ",
                "6. Reset Form ",
                "7. Exit",
            };
        }

        
    }
}

File: src/Views/SignInView.py
# Refactored SignInView.py

import os
from Credential import Credential  # Ensure Credential.py exists


class SignInView:
    def __init__(self):
        self.choice = 0
        self.menu_list = []

    def load_menu_list(self):
        """Loads the menu options."""
        self.menu_list = [
            "1. 🆔 Enter Login ID",
            "2. 🔑 Enter Password",
            "3. ✅ Login",
            "4. 📝 Sign Up",
            "5. 🔄 Forgot Password",
            "6. 🔃 Reset Form",
            "7. 🚪 Exit"
        ]

    def clear_screen(self):
        """Clears the console screen for better UI experience."""
        os.system("cls" if os.name == "nt" else "clear")

    def view(self):
        """Main SignIn menu loop."""
        self.load_menu_list()

        while True:
            self.clear_screen()
            print("\n\t🎬 ----- !!! Salam Hindusthan !!! ----- 🎬")
            print("================================================")
            print("🔑 SignIn View - CineComplex")
            print("-------------------------------------------------")
            print(f"📌 Entered Login ID: {Credential.Instance.LoginId if Credential.Instance.LoginId else 'Not entered'}")
            print(f"📌 Entered Password: {'*' * len(Credential.Instance.Password) if Credential.Instance.Password else 'Not entered'}")
            print("\n📜 Menu:")
            print("---------------")

            for instr in self.menu_list:
                print(instr)

            choice = input("🎯 Enter Your Choice: ").strip()

            if not choice.isdigit():
                print("❌ Invalid choice! Please enter a number.")
                input("🔄 Press Enter to continue...")
                continue

            self.choice = int(choice)

            if self.choice == 1:
                print("🆔 Enter Login ID: ")
                SignInView.Instance.get_sign_in_id()
            elif self.choice == 2:
                print("🔑 Enter Password: ")
                SignInView.Instance.get_sign_in_password()
            elif self.choice == 3:
                print("🔍 Authenticating...")
                authentication_result = SignInView.Instance.sign_in()
                if authentication_result.is_successful:
                    print("✅ Login Successful! Redirecting to Admin Home...")
                    AdminHomeView.Instance.View()
                else:
                    print(f"❌ {authentication_result.message}")
                    input("🔄 Press Enter to continue...")
            elif self.choice == 4:
                print("📝 Redirecting to Sign-Up...")
                SignUpFormView.Instance.View()
            elif self.choice == 5:
                print("🔄 Redirecting to Forgot Password...")
                ForgotPasswordFormView.Instance.View()
            elif self.choice == 6:
                print("🔃 Resetting form...")
                SignInView.Instance.reset_form_command()
            elif self.choice == 7:
                confirm_exit = input("⚠️ Are you sure you want to exit? (yes/no): ").strip().lower()
                if confirm_exit == "yes":
                    print("👋 Exiting... Thank you for using CineComplex!")
                    break
            else:
                print("❌ Please enter a valid choice.")

# Example usage
if __name__ == "__main__":
    sign_in_view = SignInView()
    sign_in_view.view()
